model/wire.h

```diff
@@ -51,6 +51,8 @@ namespace minsky
 
     constexpr static float handleRadius=3;
     mutable int unitsCtr=0; ///< for detecting wiring loops in units()
+  protected:  
+    mutable std::vector<std::pair<float,float>> cairoCoords; ///< contains all the internal cairo coordinates used to draw a wire
```

Comment:
IIUC, this member is only used within draw(), and draw() computes its value every time, so you could just have calcCairoCoords() return the vector of points, and do away with this mutable member.

Does anything inherit from Wire? Just wondering why you chose protected rather than private here.

Reply:
Ah sorry, I just saw the access to cairoCoords in near().

Reply:
Thanks, making it private but leaving the mutable member in. I don't foresee anything inheriting from Wire using it.